assets/blocks/course-outline/module-block/edit.js

```diff
@@ -29,24 +28,44 @@ import { useInsertLessonBlock } from './use-insert-lesson-block';
  * @param {string}   props.attributes.blockStyle  Selected block style.
  * @param {Object}   props.mainColor              Header main color.
  * @param {Object}   props.textColor              Header text color.
+ * @param {Object}   props.borderColor            Border color.
  * @param {Function} props.setAttributes          Block set attributes function.
  */
 export const EditModuleBlock = ( props ) => {
 	const {
 		clientId,
 		className,
-		attributes: { title, description },
+		attributes: { title, description, bordered, borderColorValue },
 		mainColor,
 		textColor,
 		setAttributes,
-		blockStyle,
+		name,
 	} = props;
 	const {
-		outlineAttributes: { collapsibleModules },
+		outlineAttributes: { collapsibleModules, moduleBorder },
 	} = useContext( OutlineAttributesContext ) || { outlineAttributes: {} };
 
 	useInsertLessonBlock( props );
 
+	const [ outlineClass, setOutlineClass ] = useState( null );
+
+	// setOutlineClass is called when there is an update in course style class only.
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	useEffect( () => {
+		applyParentStyle(
+			'sensei-lms/course-outline',
+			name,
+			clientId,
+			outlineClass,
+			setOutlineClass
+		);
+	} );
+
+	// Get the border setting from the parent if none is set.
+	if ( undefined === bordered ) {
```

Comment:
Considering that the purpose of that part is to make a modification, I think it'd make sense to be inside an effect.

Reply:
Do you mean `useEffect`? Why should every modification be inside an effect? Are there any benefits? 

The problem with this (apart from the added complexity) is that it will be executed after everything is rendered in the browser so there is going to be a slight delay for the border to appear. It's not a big issue but we should avoid it if it is not necessary.

Reply:
> Do you mean useEffect?

Yep!

> Why should every modification be inside an effect? Are there any benefits?

Usually, we use the function scope for only rendering stuff (including loops, conditionals, functions...), and set the hooks.

One of the benefits, is to run pieces of code only when needed (in the old times we would use the lifecycles for it, like `componentDidMount`, `componentDidUpdate`, and so on).

Another benefit is the code organization, that is easier to understand what's the component effects and what's part of the component rendering.

For this specific case, in terms of running, it wouldn't be a big difference because we'd have `useEffect` dependencies check instead of the `( undefined === bordered )` in the render. But it's more about good practices.

> it will be executed after everything is rendered in the browser so there is going to be a slight delay for the border to appear

Actually in this case, the behavior would be very similar, because the `setAttributes` is async, like state, reducers, and so on. So this would happen only after the rendering, getting the new `bordered`, only in the next render. Similar to the `useEffect` that would happen after the render and run the `setAttributes`, causing a new render.

Reply:
Ah indeed it would better and safe to have it inside an effect! Thanks for the input :) 1a1b2199af54c3b7ec652e1bb7448bc510986763